quinn/src/streams.rs

```diff
@@ -149,6 +151,27 @@ where
         conn.wake();
     }
 
+    /// Completes if/when the peer stops the stream, yielding the error code
+    pub fn stopped(&mut self) -> Stopped<'_, S> {
+        Stopped { stream: self }
+    }
+
+    #[doc(hidden)]
+    pub fn poll_stopped(&mut self, cx: &mut Context) -> Poll<VarInt> {
+        let mut conn = self.conn.lock().unwrap();
+        if self.stopped_event.is_none() {
+            let (send, recv) = oneshot::channel();
+            self.stopped_event = Some(recv);
+            conn.stopped.insert(self.stream, send);
+        }
```

Comment:
What if the stream was stopped before this function was called for the first time?

Reply:
That's a good question! I'll seek a place to store some state with `stop_reason` somewhere.